addons/account_audit_trail/models/mail_message.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from markupsafe import Markup
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class Message(models.Model):
+    _inherit = 'mail.message'
+
+    account_audit_log_preview = fields.Html(string="Description", compute="_compute_account_audit_log_preview")
+    account_audit_log_move_id = fields.Many2one(
+        comodel_name='account.move',
+        string="Journal Entry",
+        compute="_compute_account_audit_log_move_id",
+        search="_search_account_audit_log_move_id",
+    )
+    show_audit_log = fields.Boolean(compute="_compute_account_audit_log_move_id", search="_search_show_audit_log")
+
+    @api.depends('body', 'subject', 'tracking_value_ids', 'subtype_id')
+    def _compute_account_audit_log_preview(self):
+        for message in self:
+            title = message.subject or message.preview
+            tracking_value_ids = message.sudo().tracking_value_ids
+            if not title and tracking_value_ids:
+                title = _("Updated")
+            elif not title and message.subtype_id and not message.subtype_id.internal:
+                title = message.subtype_id.display_name
+            audit_log_preview = Markup("<div>%s</div>") % (title)
+            trackings = [
+                (
+                    fmt_vals['changedField'],
+                    fmt_vals['oldValue']['value'],
+                    fmt_vals['newValue']['value'],
+                ) for fmt_vals in tracking_value_ids._tracking_value_format()
+            ]
+            for field_desc, old_value, new_value in trackings:
+                audit_log_preview += Markup(
+                    "<li>%(old_value)s <i class='o_TrackingValue_separator fa fa-long-arrow-right mx-1 text-600' title='%(title)s' role='img' aria-label='%(title)s'></i>%(new_value)s (%(field)s)</li>"
+                ) % {
+                    'old_value': old_value,
+                    'new_value': new_value,
+                    'title': _("Changed"),
+                    'field': field_desc,
+                }
+            message.account_audit_log_preview = audit_log_preview
+
+    @api.depends('model', 'res_id')
+    def _compute_account_audit_log_move_id(self):
+        messages_of_account_move = self.filtered(lambda m: m.model == 'account.move' and m.res_id)
+        (self - messages_of_account_move).write({
+            'account_audit_log_move_id': False,
+            'show_audit_log': False,
+        })
```

Comment:
Don't use `write` in a compute function, it will effectively write on the record instead of just changing the cached value